From a3cb17e9cb9531414e8d91bf9c9880088f95a21f Mon Sep 17 00:00:00 2001
From: svenate <salini.venate@intel.com>
Date: Wed, 18 May 2022 22:52:02 +0530
Subject: [PATCH] WA - Add definition for pthread_rwlock_t

Added definition for pthread_rwlock_t to fix below error:
error: unknown type name 'pthread_rwlock_t'.

Signed-off-by: Vinay Kompella <vinay.kompella@intel.com>
Signed-off-by: svenate <salini.venate@intel.com>
---
 libsslsupport/openssl_support.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/libsslsupport/openssl_support.h b/libsslsupport/openssl_support.h
index 777f1be..38939fe 100644
--- a/libsslsupport/openssl_support.h
+++ b/libsslsupport/openssl_support.h
@@ -10,4 +10,12 @@ typedef UINTN size_t;
 typedef long time_t;
 typedef VOID *FILE;
 
+typedef struct {
+#if defined(__LP64__)
+  int32_t __private[14];
+#else
+  int32_t __private[10];
+#endif
+} pthread_rwlock_t;
+
 #endif 	/* _OPENSSL_SUPPORT_H_ */
-- 
2.17.1

